.github/workflows/python.yml

```diff
@@ -0,0 +1,198 @@
+name: Python
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+    paths:
+      - python/**
+  pull_request:
+    paths:
+      - python/**
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - name: Build
+        uses: PyO3/maturin-action@v1
+        with:
+          command: develop
+          args: -m python/Cargo.toml
+      - name: Install dev requirements
+        run: pip install -r python/requirements-dev.txt
+      - name: Check
+        run: mypy python && ruff check python && ruff format python --check
+      - name: Test
+        run: pytest python
+  linux:
```

Comment:
Up to you but I don't usually find it necessary to build wheels on every PR

Reply:
Agreed ... I put a `path` filter for `on` to only run when we update the Python, which _shouldn't_ be that often?